neutron/neutron.go

```diff
@@ -231,6 +233,85 @@ func (c *Client) DeleteFloatingIPV2(ipId string) error {
 	return err
 }
 
+// PortV2 describes a defined network for creating a port.
+type PortV2 struct {
+	Description    string           `json:"description,omitempty"`
+	FixedIPs       []PortFixedIPsV2 `json:"fixed_ips,omitempty"`
+	Id             string           `json:"id,omitempty"`
+	Name           string           `json:"name,omitempty"`
+	NetworkId      string           `json:"network_id"`
+	SecurityGroups []string         `json:"security_groups,omitempty"`
+	Status         string           `json:"status,omitempty"`
+	TenantId       string           `json:"tenant_id,omitempty"`
+}
+
+// PortFixedIPsV2 represents a FIxedIp with ip addresses and an associated
+// subnet id.
+type PortFixedIPsV2 struct {
+	IPAddress string `json:"ip_address"`
+	SubnetID  string `json:"subnet_id"`
+}
+
+// ListPortsV2 lists NetworkIds, names, and other details for all ports.
+func (c *Client) ListPortsV2() ([]PortV2, error) {
```

Comment:
Suggest you add Filter to this, like used for ListNetworks and ListFloatingIPs.  Might be nice later.